internal/infrastructure/middleware/metrics.go

```diff
@@ -50,6 +55,29 @@ func MetricsMiddlewareWithConfig(config *MetricsConfig) echo.MiddlewareFunc {
 				status = httpErr.Code
 			}
 			statusStr := strconv.Itoa(status)
+			headerBuf := strings.Builder{}
+			if errHeaderBuf := ctx.Request().Header.Write(&headerBuf); errHeaderBuf == nil {
+				headerSize := float64(len(headerBuf.String()))
+				config.Metrics.HTTPRequestHeaderSize.WithLabelValues(statusStr, method, path).Observe(headerSize)
+				logger.Debug("measured http_request_header_size",
+					slog.String("status", statusStr),
+					slog.String("method", method),
+					slog.String("path", path),
+					slog.Float64("size", headerSize),
+				)
+			} else {
+				logger.Warn("writing headers in string buffer",
+					slog.String("err", errHeaderBuf.Error()),
+				)
+			}
+			bodySize := float64(ctx.Request().ContentLength)
+			config.Metrics.HTTPRequestBodySize.WithLabelValues(statusStr, method, path).Observe(bodySize)
+			logger.Debug("measured http_request_body_size",
+				slog.String("status", statusStr),
+				slog.String("method", method),
+				slog.String("path", path),
+				slog.Float64("size", bodySize),
+			)
 
 			config.Metrics.HTTPRequestDuration.WithLabelValues(statusStr, method, path).Observe(time.Since(start).Seconds())
 
```

Comment:
I was thinking about the logging here. We log the body and header size but don't log the duration. What would you think about single log statement covering all?

I.e. the log output in debug mode would be more concise but having more info.

```diff
diff --git a/internal/infrastructure/middleware/metrics.go b/internal/infrastructure/middleware/metrics.go
index 202870a..b904496 100644
--- a/internal/infrastructure/middleware/metrics.go
+++ b/internal/infrastructure/middleware/metrics.go
@@ -56,15 +56,10 @@ func MetricsMiddlewareWithConfig(config *MetricsConfig) echo.MiddlewareFunc {
                        }
                        statusStr := strconv.Itoa(status)
                        headerBuf := strings.Builder{}
+                       headerSize := 0.0
                        if errHeaderBuf := ctx.Request().Header.Write(&headerBuf); errHeaderBuf == nil {
-                               headerSize := float64(len(headerBuf.String()))
+                               headerSize = float64(len(headerBuf.String()))
                                config.Metrics.HTTPRequestHeaderSize.WithLabelValues(statusStr, method, path).Observe(headerSize)
-                               logger.Debug("measured http_request_header_size",
-                                       slog.String("status", statusStr),
-                                       slog.String("method", method),
-                                       slog.String("path", path),
-                                       slog.Float64("size", headerSize),
-                               )
                        } else {
                                logger.Warn("writing headers in string buffer",
                                        slog.String("err", errHeaderBuf.Error()),
@@ -72,15 +67,19 @@ func MetricsMiddlewareWithConfig(config *MetricsConfig) echo.MiddlewareFunc {
                        }
                        bodySize := float64(ctx.Request().ContentLength)
                        config.Metrics.HTTPRequestBodySize.WithLabelValues(statusStr, method, path).Observe(bodySize)
-                       logger.Debug("measured http_request_body_size",
+
+                       durationSeconds := time.Since(start).Seconds()
+                       config.Metrics.HTTPRequestDuration.WithLabelValues(statusStr, method, path).Observe(durationSeconds)
+
+                       logger.Debug("measured http request",
                                slog.String("status", statusStr),
                                slog.String("method", method),
                                slog.String("path", path),
-                               slog.Float64("size", bodySize),
+                               slog.Float64("body_size", bodySize),
+                               slog.Float64("header_size", headerSize),
+                               slog.Float64("duration_seconds", durationSeconds),
                        )
 
-                       config.Metrics.HTTPRequestDuration.WithLabelValues(statusStr, method, path).Observe(time.Since(start).Seconds())
-
                        return err
                }
        }
```

Reply:
ok, I'd still include even the time as I think the one log statement doesn't play a big role (there might be some stuff after it anyway in the echo framework). 

But from the PoV of this PR it LGTM. So will approve.